osutil/env.go

```diff
@@ -65,51 +68,199 @@ func GetenvInt64(key string, dflt ...int64) int64 {
 	return 0
 }
 
-// EnvMap takes a list of "key=value" strings and transforms them into
-// a map.
-func EnvMap(env []string) map[string]string {
-	out := make(map[string]string, len(env))
-	for _, kv := range env {
-		l := strings.SplitN(kv, "=", 2)
-		if len(l) == 2 {
-			out[l[0]] = l[1]
-		}
+// Environment is an unordered map of key=value strings.
+//
+// Environment can be manipulated with available methods and eventually
+// converted to low-level representation necessary when executing programs.
+// This approach discourages operations that could result in duplicate
+// environment variable definitions from being constructed.
+type Environment map[string]string
+
+// NewEnvironment returns an environment with a copy of given entries.
+func NewEnvironment(entries map[string]string) Environment {
```

Comment:
this seems to be used only in tests, with the change to the Environment type definition itself, it can probably be dropped,

Enviroment{"a":....}

seems fairly natural for where it is needed

Reply:
Done